### PR TITLE
Move CLI to repo

### PR DESCRIPTION
**Pull request by jlvandenhout**
# Description of change

The CLI is mainly used to locally test documentation in external repositories. The CLI is highly dependent on styling and components provided by the Wiki repo, so it makes sense to move it into this repo, instead of mirroring any changes to the Wiki repo to the CLI repo every time.

## Type of change

- Enhancement (a non-breaking change which adds functionality)

## How the change has been tested

Local Wiki build

## Change checklist

Tick the boxes that are relevant to your changes, and delete any items that are not.

- [ ] I have followed the contribution guidelines for this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have tested the wiki locally and tested that all external links work

### CODE REVIEW

File: .github/workflows/publish.yaml [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+          file-name: ./cli/package.json
+    outputs:
+      cli: ${{ steps.cli.outputs.changed }}
+  publish-cli:
+    runs-on: ubuntu-latest
+    needs: version-changed
+    if: ${{ needs.version-changed.outputs.cli }}
+    env:
+      # Only set 'latest' tag when pushing to main
+      npm_config_tag: ${{ github.ref == 'refs/heads/main' && 'latest' || 'experimental' }}
```
Comment (jlvandenhout) [inline .github/workflows/publish.yaml:26]: Maybe we can do something like:

```suggestion
      IS_MAIN: ${{ github.ref == 'refs/heads/main' }}
      TAG: ${{ $IS_MAIN && 'latest' || 'experimental' }}
```

File: .github/workflows/publish.yaml [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+      - name: Install dependencies
+        run: |
+          yarn install
+      # Publish to NMP
+      - uses: actions/setup-node@v3
+        with:
+          registry-url: 'https://registry.npmjs.org'
+      - name: Publish to NPM
+        run: |
+          yarn workspace @iota-community/iota-wiki-cli npm publish --access public
```
Comment (jlvandenhout) [inline .github/workflows/publish.yaml:45]: And then do something like this here


```suggestion
          yarn workspace @iota-community/iota-wiki-cli npm publish --access public --tag $TAG --tolerate-republish $IS_MAIN
```

Would that work?

File: .github/workflows/publish.yaml (hunk 1 above)
Comment (Dr-Electron) [reply .github/workflows/publish.yaml:26]: Don't you think we should just be able to use `YARN_NPM_CONFIG_TAG`?

File: .github/workflows/publish.yaml (hunk 2 above)
Comment (Dr-Electron) [reply .github/workflows/publish.yaml]: Why do we want the --tolerate-republish thingy?

File: .github/workflows/publish.yaml (hunk 1 above)
Comment (Dr-Electron) [reply .github/workflows/publish.yaml:26]: Seems like this is not supported anymore in modern yarn

Comment (Dr-Electron) [reply .github/workflows/publish.yaml:26]:
```suggestion
      TAG: ${{ github.ref == 'refs/heads/main' && 'latest' || 'experimental' }}
```

File: .github/workflows/publish.yaml [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+        run: |
+          yarn workspace @iota-wiki/cli npm publish --access public --tag $TAG
+        env:
+          YARN_NPM_AUTH_TOKEN: ${{ secrets.NPM_TOKEN }}
+          YARN_NPM_REGISTRY_SERVER: 'https://registry.npmjs.org'
+      # Publish to GitHub
+      - uses: actions/setup-node@v3
+      - name: Publish to GitHub
+        run: |
+          yarn workspace @iota-wiki/cli npm publish --access public --tag $TAG
```
Comment (Dr-Electron) [inline .github/workflows/publish.yaml:52]:
```suggestion
      IS_MAIN: ${{ github.ref == 'refs/heads/main' }}
      TAG: ${{ $IS_MAIN && 'latest' || 'experimental' }}
    steps:
      - uses: actions/checkout@v3
      - uses: actions/setup-node@v3
        with:
          node-version: 16
          cache: 'yarn'
      - name: Allow modern Yarn
        run: |
          corepack enable
      - name: Install dependencies
        run: |
          yarn install
      # Publish to NMP
      - uses: actions/setup-node@v3
      - name: Publish to NPM
        run: |
          yarn workspace @iota-wiki/cli npm publish --access public --tag $TAG --tolerate-republish $IS_MAIN
        env:
          YARN_NPM_AUTH_TOKEN: ${{ secrets.NPM_TOKEN }}
          YARN_NPM_REGISTRY_SERVER: 'https://registry.npmjs.org'
      # Publish to GitHub
      - uses: actions/setup-node@v3
      - name: Publish to GitHub
        run: |
          yarn workspace @iota-wiki/cli npm publish --access public --tag $TAG --tolerate-republish $IS_MAIN
```

File: .github/workflows/publish.yaml (hunk 1 above)
Comment (Dr-Electron) [reply .github/workflows/publish.yaml]: Fixed with https://github.com/iota-wiki/iota-wiki/pull/785/commits/6f47e36368319fd1ca3e459d97e8b9639c11aa92 and https://github.com/iota-wiki/iota-wiki/pull/785/commits/2b05a17d9ce5a17c7907128bb8fd796498e8bac5

File: .github/workflows/publish.yaml (hunk 2 above)
Comment (Dr-Electron) [reply .github/workflows/publish.yaml]: Added with https://github.com/iota-wiki/iota-wiki/pull/785/commits/2b05a17d9ce5a17c7907128bb8fd796498e8bac5

**Pull request merged**